### PR TITLE
Printing: Updated mpprint to behave like pprint and defined vpretty

### LINKED ISSUE #6340: Pretty printing in Mechanics
```
When printing a "dynamicsymbol" (really just an undefined function of t) which is in a Pow, mechanics pretty printer shows the "(t)" when it should have hidden it. See below.

In [1]: from sympy.physics.mechanics import *
In [2]: q1 = dynamicsymbols('q1')
In [3]: mpprint(q1**2)
  2   
q₁ (t)
In [4]: mprint(q1**2)
q1**2
In [5]: mpprint(q1)
q₁
```

Original issue for #6340: http://code.google.com/p/sympy/issues/detail?id=3241
Original author: https://code.google.com/u/102887550923201014259/

### LINKED ISSUE #14425: physics.mechanics mpprint doesn't work with some greek letters
from sympy import diff
    from sympy.physics.mechanics import dynamicsymbols, mpprint
    mpprint(diff(dynamicsymbols('a')))
    mpprint(diff(dynamicsymbols('b')))
    mpprint(diff(dynamicsymbols('o')))
    mpprint(diff(dynamicsymbols('alpha')))
    mpprint(diff(dynamicsymbols('beta')))
    mpprint(diff(dynamicsymbols('theta')))

The first 3 lines print as expected, the characters a, b, and o with dot notation above. The latter 3 lines just print the alpha/beta/theta characters without any dot.

### PR DESCRIPTION
**Pull request by lilithxxx**
#### References to other Issues or PRs

closes #6340 , closes #14425 
#### Brief description of what is fixed or changed
```mpprint``` will behave like ```pprint``` and will print to stdout instead of returning a string. This resolve a lot of issues considering the usage of ```mpprint``` and also print the special characters correctly.

There is also a new function defined called ```vpretty``` which does return the string and this is used for the unit tests.

### CODE REVIEW

**Comment by lilithxxx** [comment]
@moorepants please review

**Comment by asmeurer** [comment]
This is a backwards compatibility break. If we do this (I'll leave that to @moorepants and the @sympy/mechanics guys to decide), we need to deprecate it somehow. 

Also, if we do make this change, there should be a function that does return the string values, which should be used in the tests (for the normal pretty printer, this is `pretty`).

File: sympy/physics/vector/tests/test_printing.py [hunk 1]
```diff
@@ -41,37 +41,75 @@ def test_vector_pretty_print():
...
-    assert unicode_vpretty(w) == uexpected
+    import sys
+    from sympy.core.compatibility import StringIO
+    result = StringIO()
+    old_stdout = sys.stdout
+    sys.stdout = result
+    try:
+        ascii_vpretty(v)
+    finally:
+        sys.stdout = old_stdout
```
Comment (asmeurer) [inline sympy/physics/vector/tests/test_printing.py]: Don't capture stdout for the tests. If we don't have a separate function that just returns the values, at the very least, just test VectorPrettyPrinter().doprint.

Comment (lilithxxx) [reply sympy/physics/vector/tests/test_printing.py]: How can I test ```VectorPrettyPrinter().doprint``` with ```use_unicode``` set to False? This is required to assert with the strings 'expected'.

**Comment by lilithxxx** [comment]
@asmeurer I have created a function called ```vpretty``` to return the str form like ```pretty``` in sympy. I don't know if that name is already taken for some other function and if so please inform me. I have also updated the tests using that function.

Should I include it in ```__all__``` to make it importable?

And should I also include a test or two for ```vpprint``` by capturing the stdouts like its done in sympy's test_pretty for ```pprint```?

**Comment by lilithxxx** [comment]
@asmeurer Please review

**Comment by lilithxxx** [comment]
@oliverlee  Do you have any suggestions on how the ```vpretty``` code should go?

**Comment by oliverlee** [comment]
`vpprint` calls `vpretty` as is similarly done with `pprint` and `pretty`: https://github.com/sympy/sympy/blob/7524499458cf114b32e3aa6e8acdb77e846111f9/sympy/printing/pretty/pretty.py#L2295-L2350

**Comment by lilithxxx** [comment]
@asmeurer I have updated the description. Please review.

File: sympy/physics/vector/tests/test_printing.py [hunk 2]
```diff
@@ -22,10 +22,10 @@
 x = alpha * (N.x | N.x) + sin(omega) * (N.y | N.z) + alpha * beta * (N.z | N.x)
 
 def ascii_vpretty(expr):
-    return vpprint(expr, use_unicode=False, wrap_line=False)
+    return vpretty(expr, use_unicode=False, wrap_line=False)
 
 def unicode_vpretty(expr):
-    return vpprint(expr, use_unicode=True, wrap_line=False)
+    return vpretty(expr, use_unicode=True, wrap_line=False)
```
Comment (moorepants) [inline sympy/physics/vector/tests/test_printing.py]: Why is a new function `vpretty()` introduced instead of fixing `vpprint()`?

**Review by moorepants — CHANGES_REQUESTED** [review]
Unit tests need to be added that check what has been fixed.

**Comment by moorepants** [comment]
@lilithxxx Can you add release notes to the first comment box as per: https://github.com/sympy/sympy/wiki/Writing-Release-Notes

**Comment by czgdp1807** [comment]
@lilithxxx Are you still working on this?

**Pull request closed without merging**